### PR TITLE
Fix parquet write decimal

### PR DESCRIPTION
**Pull request by frankgh**
Writing decimals to parquet produces the incorrect parquet file, and it cannot be read back correctly again.

### CODE REVIEW

File: server/build.gradle [hunk 1]
```diff
@@ -188,6 +188,7 @@ project('pxf-hdfs') {
         compile "org.apache.parquet:parquet-encoding:${parquetVersion}"
         compile "org.apache.parquet:parquet-generator:${parquetVersion}"
         compile "org.apache.parquet:parquet-hadoop:${parquetVersion}"
+        compile "org.apache.hive:hive-storage-api:2.4.0" // for parquet writing
```
Comment (denalex) [inline server/build.gradle]: new dependency that we ship ?

File: server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/ParquetFileAccessor.java [hunk 2]
```diff
@@ -67,15 +68,23 @@
...
     private static final CompressionCodecName DEFAULT_COMPRESSION = CompressionCodecName.SNAPPY;
 
+    public static final int PRECISION_TO_BYTE_COUNT[] = new int[38];
+
+    static {
+        for (int prec = 1; prec <= 38; prec++) {
+            // Estimated number of bytes needed.
+            PRECISION_TO_BYTE_COUNT[prec - 1] = (int)
+                    Math.ceil((Math.log(Math.pow(10, prec) - 1) / Math.log(2) + 1) / 8);
+        }
```
Comment (denalex) [inline server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/ParquetFileAccessor.java]: was this copied from somewhere ?

File: server/build.gradle (hunk 1 above)
Comment (frankgh) [reply server/build.gradle]: not really, this dependency already existed for hive

File: server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/ParquetFileAccessor.java (hunk 2 above)
Comment (frankgh) [reply server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/ParquetFileAccessor.java]: yes. I will add a comment

**Comment by frankgh** [comment]
@denalex DECIMAL/NUMERIC types that defined a precision/scale were not working. ie

```
CREATE WRITABLE EXTERNAL TABLE foo (
a NUMERIC (5,2)
) .....
```

This PR fixes NUMERIC/DECIMAL types that have a precision/scale defined

**Pull request merged**